spark-connector/src/main/java/com/datastrato/gravitino/spark/connector/SparkTransformConverter.java

```diff
@@ -0,0 +1,227 @@
+ * Spark bucket transform is corresponding to Gravitino Hash distribution without sort orders. Spark
+ * sorted bucket transform is corresponding to Gravitino Hash distribution with sort orders.
+ */
+public class SparkTransformConverter {
+
+  public static class GravitinoTransformBundles {
+    private List<Transform> partitions;
+    @Getter private Distribution distribution;
+    @Getter private SortOrder[] sortOrders;
+
+    public Transform[] getPartitions() {
+      if (partitions == null) {
+        return null;
+      }
+      return partitions.toArray(new Transform[0]);
+    }
+
+    private void addPartition(Transform partition) {
+      if (partitions == null) {
+        this.partitions = new ArrayList<>();
+      }
+      partitions.add(partition);
+    }
+
+    private void setDistribution(Distribution distribution) {
+      Preconditions.checkState(this.distribution == null, "Should only set distribution once");
+      this.distribution = distribution;
+    }
+
+    private void setSortOrders(SortOrder[] sortOrders) {
+      Preconditions.checkState(this.sortOrders == null, "Should only set sort orders once");
+      this.sortOrders = sortOrders;
+    }
+  }
+
+  public static GravitinoTransformBundles toGravitinoTransform(
+      org.apache.spark.sql.connector.expressions.Transform[] transforms) {
+    GravitinoTransformBundles bundles = new GravitinoTransformBundles();
+    Arrays.stream(transforms)
+        .forEach(
+            transform -> {
+              if (transform instanceof IdentityTransform) {
+                IdentityTransform identityTransform = (IdentityTransform) transform;
+                bundles.addPartition(
+                    Transforms.identity(identityTransform.reference().fieldNames()));
+              } else if (transform instanceof SortedBucketTransform) {
+                Pair<Distribution, SortOrder[]> pair =
+                    toGravitinoDistributionAndSortOrders((SortedBucketTransform) transform);
+                bundles.setDistribution(pair.getLeft());
+                bundles.setSortOrders(pair.getRight());
+              } else if (transform instanceof BucketTransform) {
+                BucketTransform bucketTransform = (BucketTransform) transform;
+                Distribution distribution = toGravitinoDistribution(bucketTransform);
+                bundles.setDistribution(distribution);
+              } else {
+                throw new NotSupportedException("Not support Spark transform: " + transform.name());
+              }
+            });
+
+    return bundles;
+  }
+
+  public static org.apache.spark.sql.connector.expressions.Transform[] toSparkTransform(
+      com.datastrato.gravitino.rel.expressions.transforms.Transform[] partitions,
+      Distribution distribution,
+      SortOrder[] sortOrder) {
+    List<org.apache.spark.sql.connector.expressions.Transform> sparkTransforms = new ArrayList<>();
+    if (partitions != null) {
```

Comment:
do you mean CollectionUtils.isNotEmpty()?  seems couldn't simply the code, because partition is an array not list,
```java
 public static boolean isNotEmpty(Collection coll) {}
```

Reply:
ArrayUtils.isNotEmpty()

Reply:
done